assets/js/modules/analytics/datastore/accounts.js

```diff
@@ -80,12 +84,105 @@ export const actions = {
 			type: RECEIVE_ACCOUNTS_PROPERTIES_PROFILES_FAILED,
 		};
 	},
+
+	/**
+	 * Creates a new Analytics account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} args              Argument params.
+	 * @param {string} args.accountName  Google Analytics account name.
+	 * @param {string} args.propertyName Google Analytics property name.
+	 * @param {string} args.profileName  Google Analytics profile name.
+	 * @param {string} args.timezone     Google Analytics timezone.
+	 * @return {Function} Generator function action.
+	 */
+	*createAccount( { accountName, propertyName, profileName, timezone } ) {
+		invariant( accountName, 'accountName is required.' );
+		invariant( propertyName, 'propertyName is required.' );
+		invariant( profileName, 'profileName is required.' );
+		invariant( timezone, 'timezone is required.' );
+
+		try {
+			const accountTicket = yield actions.fetchCreateAccount( { accountName, propertyName, profileName, timezone } );
+			yield {
+				payload: {},
+				type: CREATE_ACCOUNT_FINISHED,
+			};
+
+			return actions.receiveCreateAccount( { accountTicket } );
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveCreateAccountFailed( { accountName, error } );
+		}
+	},
+
+	*fetchCreateAccount( { accountName, propertyName, profileName, timezone } ) {
+		yield {
+			payload: {},
+			type: CREATE_ACCOUNT_STARTED,
+		};
+
+		return {
+			payload: { accountName, propertyName, profileName, timezone },
+			type: FETCH_CREATE_ACCOUNT,
+		};
```

Comment:
Not a big deal, but it feels cleaner to have the `CREATE_ACCOUNT_FINISHED` action creator part in here too, instead of separately above. You should be able to do `response = yield ...` on the above object, then add the `yield` for `CREATE_ACCOUNT_FINISHED`, and at the very end `return response;`. See e.g. https://github.com/google/site-kit-wp/pull/1311/files#diff-0794af76964966f1028087bd15b64452R57-R60, https://github.com/google/site-kit-wp/pull/1311/files#diff-0794af76964966f1028087bd15b64452R71-R74, and https://github.com/google/site-kit-wp/pull/1311/files#diff-0794af76964966f1028087bd15b64452R87 for a similar code example.

Reply:
ok, i'll try it that way.

Reply:
moved the try catch in https://github.com/google/site-kit-wp/pull/1390/commits/26e7f06b0ee2bce7eab7540d487c41a0a47cc2ad